diff --git a/CMakeLists.txt b/CMakeLists.txt
index 6b4d4d8..b3ea1dd 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -74,6 +74,7 @@ set(FAAD_DEFINES
   HAVE_STRINGS_H=1
   HAVE_SYS_STAT_H=1
   HAVE_SYS_TYPES_H=1
+  HAVE_LRINTF=1
   STDC_HEADERS=1
   PACKAGE_VERSION=\"${FAAD_VERSION}\"
 )
@@ -220,7 +221,7 @@ endif()
 
 # Installation
 
-if(NOT FAAD_BUNDLED_MODE AND NOT MSVC)
+if(NOT FAAD_BUNDLED_MODE)
   install(FILES "${CMAKE_CURRENT_BINARY_DIR}/faad2.pc"
     DESTINATION "${CMAKE_INSTALL_LIBDIR}/pkgconfig")
 
