diff --git a/meson.build b/meson.build
index 7c67c94..c732002 100644
--- a/meson.build
+++ b/meson.build
@@ -180,7 +180,6 @@ if libproxy_dep.found() or gsettings_desktop_schemas_dep.found()
     subdir('proxy/gnome')
   endif
 
-  subdir('proxy/tests')
 endif
 
 subdir('tls/base')
@@ -197,8 +196,6 @@ if backends.contains('openssl')
   subdir('tls/openssl')
 endif
 
-subdir('tls/tests')
-
 # Will automatically pick it up from the cross file if defined
 gio_querymodules = find_program('gio-querymodules', required : false)
 if gio_querymodules.found()
