diff --git a/proxy/meson.build b/proxy/meson.build
index c638452..e989a2a 100644
--- a/proxy/meson.build
+++ b/proxy/meson.build
@@ -11,5 +11,3 @@ endif
 if not ['windows'].contains(host_system)
   subdir('environment')
 endif
-
-subdir('tests')
diff --git a/tls/meson.build b/tls/meson.build
index e76e766..b8103cd 100644
--- a/tls/meson.build
+++ b/tls/meson.build
@@ -13,5 +13,3 @@ if backends.contains('openssl')
 
   subdir('openssl')
 endif
-
-subdir('tests')
