diff --git a/libsoup/soup-http2-utils.h b/libsoup/soup-http2-utils.h
index f3a7c73..a30ca83 100644
--- a/libsoup/soup-http2-utils.h
+++ b/libsoup/soup-http2-utils.h
@@ -4,6 +4,13 @@
 
 #pragma once
 
+#ifdef _MSC_VER
+#include <BaseTsd.h>
+typedef SSIZE_T ssize_t;
+#else
+#include <unistd.h>
+#endif
+
 #include <nghttp2/nghttp2.h>
 
 #define NGCHECK(stm)                                                                             \
